src/python/zquantum/core/circuits/_matrices.py

```diff
@@ -117,13 +117,10 @@ def phase_matrix(angle):
 
 def u3_matrix(theta, phi, lambda_):
     """Based on
-    https://github.com/quantumlib/Cirq/blob/292080453e22e91dc5658a0cfa5043539944a950/cirq/circuits/qasm_output.py#L70
+    https://qiskit.org/documentation/stubs/qiskit.circuit.library.U3Gate.html
+    Please note that this formulation introduces a global phase
     """
-    return (
-        rz_matrix(phi % (2 * sympy.pi))
-        * ry_matrix(theta % (2 * sympy.pi))
-        * rz_matrix(lambda_ % (2 * sympy.pi))
-    )
+    return rz_matrix(phi) * ry_matrix(theta) * rz_matrix(lambda_)
```

Comment:
The Qiskit formula features 5 gates, whereas ours uses 3. Are these two formulas still matching?

Reply:
Qiskit sandwiches an RZ between two RX gates with constant pi/2 rotations. This is equivalent to having a single RY gate with the rotation of the RZ. It's a nice property of rotational gates.

Reply:
That's also how Cirq implements it internally [here](https://github.com/quantumlib/Cirq/blob/292080453e22e91dc5658a0cfa5043539944a950/cirq/circuits/qasm_output.py#L70)